documentation/src/main/asciidoc/user_guide/cloud.adoc

```diff
@@ -62,3 +62,23 @@ Finally you can configure your JGroups instances to use a shared storage to exch
 ==== JDBC_PING
 A similar approach to S3_PING, but using a JDBC connection to a shared database. On EC2 that is quite easy using Amazon RDS. See the link:$$http://community.jboss.org/wiki/JDBCPING$$[JDBC_PING Wiki page] for details. 
 
+==  Using Kubernetes and OpenShift Rolling Updates
+
+TBD!!!
+
+==  Rolling upgrades
+
+=== Kubernetes and OpenShift
+
+For both Kubernetes and OpenShift, the Rolling Upgrade procedure is almost the same. It is based on a standard <<_Rolling_chapter,Rolling Upgrade procedure>> with small changes.
+
+.Key differences when upgrading using OpenShift/Kubernetes are:
+* When forming a new cluster, make sure you use labels. Labels can be used by link:$$https://github.com/jgroups-extras/jgroups-kubernetes$$[Kubernetes PING Protocol] (see `OPENSHIFT_KUBE_PING_LABELS` environment variable) and they allow controlling which nodes are assigned to which clusters.
+* Depending on configuration, it is a good practice to use link:$$https://docs.openshift.org/latest/architecture/core_concepts/routes.html$$[OpenShift Routes] or link:$$http://kubernetes.io/docs/user-guide/ingress$$[Kubernetes Ingress API] to expose services to the clients. During the upgrade the Route (or Ingress) used by the clients can be altered to point to the new cluster.
+* Invoking CLI commands can be done by using Kubernetes (`kubectl exec`) or OpenShift clients (`oc exec`). Here is an example: `oc exec <POD_NAME> -- '/opt/jboss/infinispan-server/bin/ispn-cli.sh' '-c' '--controller=$(hostname -i):9990' '/subsystem=datagrid-infinispan/cache-container=clustered/distributed-cache=default:disconnect-source(migrator-name=hotrod)'`
```

Comment:
Hmmm, `hostname -i` gives `fe80::xxxx:xxxx:xxxx:xxxx%wlp4s0 10.x.x.x 192.168.x.x` on my machine. Are OpenShift hosts guaranteed to have just 1 IPv4 address?